api/client/proto/authservice.proto

```diff
@@ -977,6 +1019,89 @@ message DeleteWindowsDesktopRequest {
     string Name = 1 [ (gogoproto.jsontag) = "name" ];
 }
 
+// ChangePasswordWithTokenRequest defines a request to change one or more of a user's auth
+// credential.
+message ChangePasswordWithTokenRequest {
+    string TokenID = 1;
+    // SecondFactorToken is an otp value.
+    string SecondFactorToken = 2;
+    // Password is user's password.
+    bytes Password = 3;
+    // U2FRegisterResponse is U2F registration challenge response.
+    U2FRegisterResponse U2FRegisterResponse = 4;
```

Comment:
Are the register response/new device mandatory? If not, I'd consider adding both to their own message, so it's clearer that all of this may absent.

Reply:
No it's not mandatory. I used `MFARegisterResponse` to combine secondFactorToken and u2fRegisterResponse (i think this is what you meant). I removed the `DeviceName` field b/c our UI doesn't allow users to provide their own device name on signup/reset forms (yet)